app/src/test/java/me/hufman/androidautoidrive/music/SpotifyWebApiTest.kt

```diff
@@ -1309,6 +1309,24 @@ class SpotifyWebApiTest {
 		verify(clientPlaylistApi).replaceClientPlaylistPlayables(playlistId, PlayableUri(songs[0].mediaId!!), PlayableUri(songs[1].mediaId!!))
 	}
 
+	@Test
+	fun testReplacePlaylistSongs_LargeNumberOfSongs() = runBlocking {
+		val playlistId = "playlistId"
+		val songs = (0..120).map { MusicMetadata("mediaId$it", it.toLong(), true, false, 2, null, "coverArtUri$it", null, "Artist $it", "Album $it", "Title $it") }
+
+		val clientPlaylistApi: ClientPlaylistApi = mock()
+		whenever(clientPlaylistApi.replaceClientPlaylistPlayables(playlistId)).doAnswer {  }
+		whenever(clientPlaylistApi.addPlayablesToClientPlaylist(eq(playlistId), any(), position = isNull())).doAnswer {  }
+
+		val webApi: SpotifyClientApi = mock()
+		whenever(webApi.playlists).thenReturn(clientPlaylistApi)
+		FieldSetter.setField(spotifyWebApi, spotifyWebApi::class.java.getDeclaredField("webApi"), webApi)
+
+		spotifyWebApi.replacePlaylistSongs(playlistId, songs)
+
+		verify(clientPlaylistApi).replaceClientPlaylistPlayables(playlistId)
+	}
```

Comment:
This should also verify that it calls `addPlayablesToClientPlaylist(eq(playlistId), any())` twice, I think

Reply:
Ah you're right, added that verify :)